### PR TITLE
Use Flutter's analysis_options file + fix all issues

### PR DESCRIPTION
**Pull request by DanTup**
I also found a few URLs where we download files from Flutter that was still using `master`, so I pulled this into a variable and set it to `dev`. To pass some lints, I also had to remove our dummy `@immutable`/`@required` attributes in `stubs` and reference the real ones.

### CODE REVIEW

File: tool/grind.dart [hunk 1]
```diff
@@ -6,38 +6,68 @@ import 'dart:convert';
...
 
 import 'common.dart';
 
-main(List<String> args) => grind(args);
+Future<void> main(List<String> args) => grind(args);
 
 @DefaultTask()
-@Depends(colors, icons, catalog, version)
-generate() => null;
+@Depends(analysisOptions, colors, icons, catalog)
```
Comment (devoncarew) [inline tool/grind.dart:16]: Is removing `version` intentional?

File: tool/icons/update_icons.dart [hunk 2]
```diff
@@ -5,20 +5,22 @@
...
-const outputFolder = 'resources/icons';
+import '../common.dart';
 
-void main() async {
+const String outputFolder = 'resources/icons';
+
+Future<void> main() async {
   // download material/icons.dart and cupertino/icons.dart
-  String materialData =
+  final String materialData =
```
Comment (devoncarew) [inline tool/icons/update_icons.dart:14]: Can you add a TODO: here to instead copy from the local flutter sdk?

File: tool/grind.dart (hunk 1 above)
Comment (DanTup) [reply tool/grind.dart:16]: Yep!

(No 😞  bad conflict resolution). Fixed!

File: tool/icons/update_icons.dart (hunk 2 above)
Comment (DanTup) [reply tool/icons/update_icons.dart:14]: Oh yeah - there's a few places we're downloading that we should now pull from the local one. I've added TODOs to each of them. Thanks!

**Comment by DanTup** [comment]
Actions still don't seem to be working, but I don't see why... pull_request is listed here:

https://github.com/flutter/tools_metadata/blob/master/.github/workflows/build.yaml#L3

Which matches the same given here:

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-a-list-of-events

🤔

**Comment by DanTup** [comment]
Actually, something seems broken... The latest changesets I've pushed are not showing here (they do show at https://github.com/DanTup/tools_metadata/commits/update-analysis-options), so maybe GitHub is having issues.

**Comment by devoncarew** [comment]
> Actually, something seems broken... The latest changesets I've pushed are not showing here (they do show at https://github.com/DanTup/tools_metadata/commits/update-analysis-options), so maybe GitHub is having issues.

Hmm - perhaps to do with the change from a private to a public repo? It happened over the course of this PR. You could try re-creating the PR.

**Comment by DanTup** [comment]
Oh yeah - looks like my fork is no longer considered a fork. I'll bin my repo and re-fork it shortly and try again.

**Comment by devoncarew** [comment]
I hit this as well - I deleted my fork and re-forked, and that worked.

**Comment by DanTup** [comment]
Re-opened as #12, which includes the new changesets addressing the comments above.

**Pull request closed without merging**